internal/controllers/clusterextension_controller.go

```diff
@@ -119,14 +119,25 @@ func checkForUnexpectedFieldChange(a, b ocv1alpha1.ClusterExtension) bool {
 //
 //nolint:unparam
 func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alpha1.ClusterExtension) (ctrl.Result, error) {
+	l := log.FromContext(ctx).WithName("operator-controller")
+
+	// Don't do anything if Paused
+	ext.Status.Paused = ext.Spec.Paused
+	if ext.Spec.Paused {
+		l.Info("resource is paused", "name", ext.GetName(), "namespace", ext.GetNamespace())
+		return ctrl.Result{}, nil
+	}
+
 	// Lookup the bundle that corresponds to the ClusterExtension's desired package.
 	bundle, err := r.resolve(ctx, ext)
 	if err != nil {
-		ext.Status.InstalledBundle = nil
-		setInstalledStatusConditionUnknown(&ext.Status.Conditions, "installation has not been attempted as resolution failed", ext.GetGeneration())
+		if c := apimeta.FindStatusCondition(ext.Status.Conditions, ocv1alpha1.TypeInstalled); c == nil {
```

Comment:
Do we need this additional check for finding status condition? _IIRC_ Installed status is added by default. I see that we are not doing this for bundle validation or unpack, so to maintain consistency, it's better to rather remove this check here, and explicitly set installed to nil regardless.

Reply:
We only need the status check if we want to preserve the installed condition from the currently installed thing in case of a resolution failure to indicate that the installed version is healthy even if there's an error on the Extension/ClusterExtension. We probably don't need this check once we have a separate healthy condition the users could look at instead.